Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -185,6 +186,77 @@ JS_DEFINE_NATIVE_FUNCTION(StringPrototype::starts_with)
     return Value(string.substring(start, search_string_length) == search_string);
 }
 
+static bool is_regexp(VM& vm, GlobalObject& global_object, Value possible_regexp)
+{
+    if (!possible_regexp.is_object())
+        return false;
+
+    auto matcher = possible_regexp.as_object().get(vm.well_known_symbol_match());
+    if (vm.exception())
+        return false;
+    if (!matcher.is_empty() && !matcher.is_undefined()) {
+        return !matcher.to_boolean();
+    }
+
+    auto regexp_as_object = possible_regexp.to_object(global_object);
+    if (vm.exception())
+        return false;
+    if (regexp_as_object->is_regexp_object()) {
+        return true;
+    }
+
+    return false;
+}
+
+JS_DEFINE_NATIVE_FUNCTION(StringPrototype::ends_with)
+{
+    auto string = ak_string_from(vm, global_object);
+    if (string.is_null())
+        return {};
+
+    auto search_string_value = vm.argument(0);
+    if (vm.exception())
+        return {};
+
+    bool search_is_regexp = is_regexp(vm, global_object, search_string_value);
+    if (vm.exception())
+        return {};
+    if (search_is_regexp) {
+        vm.throw_exception<TypeError>(global_object, ErrorType::IsNotA, "searchString", "string, but a regular expression");
+        return {};
+    }
+
+    auto search_string = vm.argument(0).to_string(global_object);
+    if (vm.exception())
+        return {};
+
+    auto string_length = string.length();
+    auto search_string_length = search_string.length();
+
+    size_t pos = string_length;
+
+    auto end_position_value = vm.argument(1);
+    if (!end_position_value.is_undefined()) {
+        auto number = end_position_value.to_number(global_object);
+        if (vm.exception())
+            return {};
+        double pos_as_double = number.to_integer_or_infinity(global_object);
+        if (vm.exception())
+            return {};
+        pos = clamp(pos_as_double, static_cast<double>(0), static_cast<double>(string_length));
+    }
+
+    if (search_string_length == 0) {
+        return Value(true);
+    }
+    if (pos < search_string_length) {
+        return Value(false);
+    }
```

Comment:
Not mandatory, but you can skip the curly braces here:

```suggestion
    if (search_string_length == 0)
        return Value(true);
    if (pos < search_string_length)
        return Value(false);
```